server/routes/charts.js

```diff
@@ -185,14 +185,26 @@ router.route('/charts/:chart_id')
           }
         }))
       const chartVariableColors = fieldValues.map(({ fieldLabelValueMap: { color }}) => color)
+      const groupTargetValuesByFieldValues = fieldValues
+        .reduce((currentGroup, nextGroup) => {
+          currentGroup[nextGroup.fieldLabelValueMap.label] = nextGroup.fieldLabelValueMap
+            .targetValues
+              .reduce((currentSites, nextSites) => {
+                currentSites[nextSites.site] = { value: nextSites.value }
+
+                return currentSites
+              }, {})
+          return currentGroup
+        }, {})
```

Comment:
I'm having a tough time understanding this, can you walk me through what we're building here?

Reply:
For this part I'm grouping the values, sites and the sites targets for that value so that I can build the template string when a user is hovering over the graph. 
`
 {
    valueA: {
       aSite: aSiteTargetValue,
       bSite: bSiteTargetValue
    } ,
    valueB: {
       aSite: aSiteTargetValue,
       bSite: bSiteTargetValue
    } 
 }
`